docs/release_notes/v0.4.0.md

```diff
@@ -0,0 +1,267 @@
+
+**Fixed** Reminder not fetching state correctly
+
+**Updated** Build for Dotnet Core 3.1 LTS
+
+**Added** InvokeClient and InvokeHttpClient implementations (Thanks @jeastham1993)
+
+**Updated** Localhost to 127.0.0.1
+
+**Updated** Actors samples to show timers and reminders (Thanks @pacodelacruz)
+
+**Fixed** Integration tests (Thanks @ryannowak)
+
+**Fixed** Actor reminders corrupt state (Thanks @chgeuer)
+
+**Added** Support for delete operations (Thanks @ryannowak)
+
+**Updated** ASP.NET core sample docs to include commands for windows
+
+**Fixed** URL generation when BaseAddress is used (Thanks @ryannowak)
+
+**Updated** Readme for Actor Sample to clarify client invocation
+
+### Java SDK
+
+**Added** Actor support
+
+**Added** Support for bindings
+
+**Added** Support for pub/sub
+
+**Added** Support for state management
+
+**Added** Support for service invocation
+
+**Added** Unit tests
+
+**Added** Integration tests
+
+**Added** Custom serialization support
+
+**Removed** Maven dependency to run dapr for examples (Thanks @Rinita-De)
+
+### Samples
+
+**Updated** All samples to account for multiple state stores API changes
+
+### Docs
+
+**Added** CloudState How-To
+
+**Added** Azure Table Storage How-To (Thanks @aloneguid)
+
+**Added** Hashicorp Vault How-To
+
+**Added** mTLS How-To
+
+**Added** Hazalcast How-To
+
+**Added** SQL Server How-To (Thanks @fbeltrao)
+
+**Added** Couchbase How-To
+
+**Added** Aerospike How-To
+
+**Added** Java SDK docs
+
+**Improved** Event Hubs binding spec
+
+**Updated** docs to comply with Microsoft Doc Authoring Pack (Thanks @LaurentLesle)
+
+**Updated** Docs to account for multiple state stores API changes
+
+**Added** IntelliJ debugging guide
+
+**Added** Azure SignarlR binding docs (Thanks @fbeltrao)
+
+**Updated** Helm 2 to Helm 3
+
+**Updated** Diagrams
+
+
+## Upgrading to Dapr 0.4.0
+
+If you're upgrading from an older version of Dapr to 0.4.0, follow the steps here to ensure a smooth upgrade. You know, the one where you don't get red errors on the terminal.. we all hate that, right?
+
+### Local Machine / Self-hosted
+
+Uninstall Dapr using the CLI you currently have:
+
+```
+dapr uninstall --all
+```
+
+Next, get the latest CLI following [these](https://github.com/dapr/cli#installing-dapr-cli) instructions, or alternatively download the latest and greatest release from [here](https://github.com/dapr/cli/releases) and put the `dapr` binary in your PATH.
+
+Once you have downloaded the CLI, run:
+
+```
+dapr init
+```
+
+Wait for the update to finish, and you're good to go!
+
+### Kubernetes
+
+Download the latest CLI release as outlined above in the `Local Machine / Self-hosted` section.
+
+If you previously installed Dapr on your Kubernetes cluster using the Dapr CLI, run:
+
+```
+dapr uninstall --kubernetes
+```
+
+It's fine to ignore any errors that might show up.
+
+If you previously installed Dapr using __Helm 2.X__:
+
+```
+helm del --purge dapr
+```
+
```

Comment:
I think we should add this link - https://helm.sh/blog/migrate-from-helm-v2-to-helm-v3/

Also we may need to guide users to clean up all apps(such as redis, kafka) which helm v2 installed.

Reply:
Helm 2 to 3 migration is beyond the scope for the release notes..

Reply:
migration is out-of-scope, but I think we need to provide the instruction for how to remove apps and tiller which helm v2 installs. I messed up my cluster because I did not delete all helm apps and tiller. @amanbha might have the similar problems.

cc/ @amanbha @msfussell

Reply:
Can you a  link to the update state store API here https://github.com/dapr/docs/blob/master/reference/api/state.md in the breaking change section

Reply:
^ @msfussell done, @youngbupark I put a note for the Helm upgrade link